### ISSUE TITLE
cannot invoke npm/npx from WSL shell. Error: Cannot find module.

### ISSUE THREAD

**Issue by Animeshz** [npm/cli#3248]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Following error is thrown:

```bash
    /mnt/d/Personal/TempDir ·································································· 3s   09:34:23 PM 
❯ npm -v
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'D:\mnt\c\Program Files\nodejs\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

### Expected Behavior

The path for finding the js file `npm-cli.js` should be `C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js`, instead of `D:\mnt\c\Program Files\nodejs\node_modules\npm\bin\npm-cli.js`.

Same goes for npx, which tries to find `npx-cli.js` in the same location.

### Steps To Reproduce

1. Install NodeJS as usual
2. Install WSL2 with a linux distro
3. Invoke `npm` or `npx` from the shell, as for example `npm -v`

### Environment

- OS: Windows 10 build 19042.985 (20H2) with WSL2 Ubuntu
- Node: v14.15.4
- npm: 7.13.0

**Comment by Animeshz** [comment]
Seems like the same error happens if we run this, if I go further I would guess there's an issue on WSL itself.

```bash
❯ node "/mnt/c/Program Files/nodejs/node_modules/npm/bin/npm-cli.js"
```

Npm seem to find correct paths as I tried to print things in between of the `bin/npm`.sl

**Comment by Animeshz** [comment]
Looks like its already fixed at e69be2ac5c35e985732e2baa00b70d39332e4b9f yet is not pushed to version 7.13.0 🤔 I just copy pasted the file for both `bin/npm` and `bin/npx` for the fix.

**Comment by ljharb** [comment]
@Animeshz that commit is included in npm v7.6.1 and higher, including v7.13.0, as you can see in the tags list in the commit page header.

**Comment by Animeshz** [comment]
@ljharb Its correct I guess, but `npm install -g npm` did not updated those files 👀 (updating from 6.x)
